internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -1134,6 +1084,67 @@ func (r *Reconciler) reconcileRestoreJob(ctx context.Context,
 	return errors.WithStack(r.apply(ctx, restoreJob))
 }
 
+func generateRestoreJobIntent(_ context.Context,
```

Comment:
:question: Just curious about the `context` here, e.g. do you see a need to at some point need to pass in a real `context`?

Reply:
Was really just following the pattern. Can take a look at removing